airflow/decorators/base.py

```diff
@@ -208,16 +209,16 @@ def __init__(
         self.op_kwargs = op_kwargs
         super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
 
-    def execute(self, context: Context):
+    def execute(self, context: Context) -> RV:
         # todo make this more generic (move to prepare_lineage) so it deals with non taskflow operators
         #  as well
         for arg in chain(self.op_args, self.op_kwargs.values()):
             if isinstance(arg, Dataset):
                 self.inlets.append(arg)
-        return_value = super().execute(context)
+        return_value: RV = super().execute(context)
```

Comment:
Not sure if you can do this, CI will tell.

Reply:
If we use a `TypeVar` bound to the variables listed in my comment below, we could maybe use the following instead:
`return_value: Dataset | list | dict = super().execute(context)`